backend/src/main/kotlin/org/darkSolace/muse/securityModule/model/UserDetails.kt

```diff
@@ -0,0 +1,29 @@
+package org.darkSolace.muse.securityModule.model
+
+import org.darkSolace.muse.userModule.model.Role
+import org.darkSolace.muse.userModule.model.User
+import org.springframework.security.core.GrantedAuthority
+import org.springframework.security.core.authority.SimpleGrantedAuthority
+import org.springframework.security.core.userdetails.UserDetails
+
+
+/**
+ * [UserDetails] model to pass required information to the [org.darkSolace.muse.securityModule.service.AuthTokenFilter]
+ */
+class UserDetails(val user: User) : UserDetails {
+    override fun getAuthorities(): Collection<GrantedAuthority> {
+        return listOf(SimpleGrantedAuthority(user.role.name))
+    }
+
+    override fun getPassword() = user.password
+
+    override fun getUsername() = user.username
+
+    override fun isAccountNonExpired() = user.role != Role.SUSPENDED
+
+    override fun isAccountNonLocked() = user.role != Role.SUSPENDED
+
+    override fun isCredentialsNonExpired() = user.role != Role.SUSPENDED
+
+    override fun isEnabled() = user.role != Role.SUSPENDED
```

Comment:
why 4 functions to return the same thing?

Reply:
those functions are defined and required by the interface. It would allow us to handle user states in different ways, e.g. handle "Credentials expired" different than "Account locked" - as we don't use this functionality only a check for "SUSPENDED" is made.
But now I remember that suspended users should be handled in a special way and we currently don't plan to have locked accounts.
So we could either return `true` for each of those checks (existing accounts are always allowed to log in) or we could introduce a role `LOCKED` to disable but not delete user accounts.

Reply:
Prepared a change which includes:
- Interface required methods mentioned above return all true - all User Accounts are "active" all the time
- If a suspended user tries to log in, instead of HTTP 200 (OK) or HTTP 401 (UNAUTHORIZED), HTTP 307 (TEMPORARY REDIRECT) without any authorization token is handed back to the caller. A (to be defined - currently "/suspended/<username>") URL to the "Code of Conduct reminder page" is set as the "Location" header in the response. The front-end could use this information to redirect the user to the appropriate page.

A commit is prepared and could be added to this PR on approval.

Reply:
HTTP 200